### PR TITLE
Add new form field to display read-only newsletter description/summary fields in between form-input fields

### PR DESCRIPTION
**Pull request by svenow**
- description fields are read-only and do only display text
- description fields can be arbitrarily positioned within the form via their weight value 
- an empty _fieldset_ is used as form-field
- the _description_ attribute of the fieldset is used to display the read-only text
  - the attribute wont introduce any additional color styling or scrollbars
  - the attribute allows multiple lines that are automatically adjusted to form width, without being cut.

*systopia-reference: 24994*

### CODE REVIEW

File: src/Form/SubscriptionForm.php [hunk 1]
```diff
@@ -83,7 +83,29 @@ public function buildForm(array $form, FormStateInterface $form_state, stdClass
     );
 
     // Build form according to received configuration:
-    // Add contact fields.
```
Comment (jensschuppe) [inline src/Form/SubscriptionForm.php:86]: This comment still makes sense …

File: src/Form/SubscriptionForm.php [hunk 2]
```diff
@@ -83,7 +83,29 @@ public function buildForm(array $form, FormStateInterface $form_state, stdClass
...
+      $is_active = isset($contact_form_description_data['active']) && $contact_form_description_data['active'];
+      $has_description = isset($contact_form_description_data['description']) && strlen($contact_form_description_data['description']) > 0;
+
+      if ($is_active && $has_description) {
+        $has_weight = isset($contact_form_description_data['weight'])
+                           && filter_var($contact_form_description_data['weight'], FILTER_VALIDATE_INT) !== FALSE;
+        $weight = $has_weight ? $contact_form_description_data['weight'] : 0;
+
+        $form['contact_fields'][$contact_form_description_name] = array(
+          '#type' => 'fieldset',
```
Comment (jensschuppe) [inline src/Form/SubscriptionForm.php]: Let's use [`markup`](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Render%21Markup.php/class/Markup) instead.

File: src/Form/SubscriptionForm.php [hunk 3]
```diff
@@ -83,7 +83,29 @@ public function buildForm(array $form, FormStateInterface $form_state, stdClass
     );
 
     // Build form according to received configuration:
-    // Add contact fields.
+
+    // Add form description fields as empty fieldsets.
+    // - Only use the #description attribute of fieldsets in order to show unformated,
+    //   multiline text without scrollbars or color formating.
+    // - #description text is automatically scalled to window width.
+    foreach ($profile->contact_form_descriptions as $contact_form_description_name => $contact_form_description_data) {
```
Comment (jensschuppe) [inline src/Form/SubscriptionForm.php]: Let's be backwards-compatible by defaulting to an empty array:
```php
foreach ($profile->contact_form_descriptions ?? [] as $contact_form_description_name => $contact_form_description_data) {
```

File: src/Form/SubscriptionForm.php [hunk 4]
```diff
@@ -83,7 +83,29 @@ public function buildForm(array $form, FormStateInterface $form_state, stdClass
...
+      $is_active = isset($contact_form_description_data['active']) && $contact_form_description_data['active'];
+      $has_description = isset($contact_form_description_data['description']) && strlen($contact_form_description_data['description']) > 0;
+
+      if ($is_active && $has_description) {
```
Comment (jensschuppe) [inline src/Form/SubscriptionForm.php]: Suggestion to shorten this to:
```php
if (
  (bool) ($contact_form_description_data['active'] ?? FALSE)
  && '' !== $contact_form_description_data['description'] ?? ''
) {
```

**Review by jensschuppe — CHANGES_REQUESTED** [review]

File: src/Form/SubscriptionForm.php (hunk 2 above)
Comment (svenow) [reply src/Form/SubscriptionForm.php]: Probably [html_entity_decode(](https://www.php.net/manual/en/function.html-entity-decode.php)$contact_form_description_data['description']) should be used.

If plain html-tags are entered into the description field on the civicrm-side (via _de.systopia.newsletter_), all those tags will be converted to their html-entity version on the CiviCrm side. Received html-entities on the Drupal side are not converted back when using a `markup` field.
  
If `html_entity_decode` is being called before assigning the description string to the _#markup_ field, then the plain html-tags would be recovered and then rendered correctly in the form:

`'#markup' => html_entity_decode($contact_form_description_data['description'])`
`'#markup' => Markup::create(html_entity_decode($contact_form_description_data['description']))`

Testing indicates that both versions produce the same correctly rendered output when receiving html-tags like  &lt;strong&gt; or  &lt;em&gt; from the civicrm-side.

`html_entity_decode` uses **utf-8** (if not specified otherwise in php.ini) as default for character conversion and has a default flag-argument set to `ENT_QUOTES | ENT_SUBSTITUTE | ENT_HTML401`

Comment (jensschuppe) [reply src/Form/SubscriptionForm.php]: The API should rather send raw HTML (which requires adjusting the HTML filtering in the CiviCRM extension), so that the Drupal module does not have to decode.

Also, ~~`Markup::create()` is the preferred method~~ **not** using `Markup::create()` is the preferred method, as it ~~implies~~ bypasses XSS filtering of raw HTML in Drupal (e.g. stripping `<script>` tags etc.).

**Review by jensschuppe — CHANGES_REQUESTED** [review]
I un-resolved some of my earlier review's points that are still valid.

**Review by jensschuppe — APPROVED** [review]
